src/pymorize/cmorizer.py

```diff
@@ -257,6 +309,46 @@ def _check_is_subperiod(self):
         if errors:
             for err in errors:
                 logger.error(err)
+            raise errors[0]
+
+    def _check_units(self):
+        def is_unit_scalar(value):
+            if value is None:
+                return True
```

Comment:
This is not very precise: if I use `is_unit_scalar(None)` then I'll get `True`, where it should be returning `False`

Reply:
I am changing this whole block so no action from you is required, @siligam

Reply:
```suggestion
                return False
```